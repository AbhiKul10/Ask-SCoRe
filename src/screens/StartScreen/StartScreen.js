import React, { Component } from 'react';
import { View, Text } from 'react-native';
import 'react-native-paper';
import LoginSignupScreen from '../LoginSignupScreen/LoginSignupScreen';
import { Spinner, Button } from '../../components/common';
import { firebase } from '../../config/config';
import HomeScreen from '../HomeScreen/HomeScreen';

class StartScreen extends Component { 
    state = {loggedIn : null};

    componentDidMount() {
         firebase.auth().onAuthStateChanged((user) => {
            if (user){
                this.setState({ loggedIn: true });
            } else {
                this.setState({ loggedIn: false });
            }
        });
    }

    renderContent() {
        switch(this.state.loggedIn){
            case true :
                return( 
                    <>
                        <HomeScreen />
                    </>
                    )
            case false : 
                return <LoginSignupScreen />;
            default : 
            return (
                <View >
                    <View>
                    <Spinner size="large" style={{justifyContent:'center'}}/>
                    </View>
                </View>    
            )
        }
    }
    
    render() {
        return(
            <>
            {this.renderContent()}
            </>
        );
    }
}

export default StartScreen;