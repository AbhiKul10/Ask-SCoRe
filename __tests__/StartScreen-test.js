import React from "react";
import StartScreen from "../src/screens/StartScreen/StartScreen";

import renderer from "react-test-renderer";

jest.useFakeTimers();

test("renders Login Component correctly", async () => {
  const tree = renderer.create(<StartScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
