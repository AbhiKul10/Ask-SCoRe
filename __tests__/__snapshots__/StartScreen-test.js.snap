// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login Component correctly 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator
        animating={true}
        color="#999999"
        hidesWhenStopped={true}
        size="large"
      />
    </View>
  </View>
</View>
`;
